### PR TITLE
Update the Terraform prefix to include both env.CHANGE_ID and BUILD_ID

### PR DESCRIPTION
**Pull request by rtyler**
I previously forgot that each new pull request is like a whole new Pipeline
whose build numbers start at 1.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -30,7 +30,7 @@ else if (env.CHANGE_ID) {
     /* When handling pull requests, ensure everything is denoted by the pull
      * request
      */
-    tfPrefix = "pr${env.BUILD_NUMBER}"
+    tfPrefix = "pr${env.CHANGE_ID}${env.BUILD_NUMBER}"
```
Comment (olblak) [inline Jenkinsfile]: What is 'env.CHANGED_ID' for?
I am looking for definition [here](https://wiki.jenkins-ci.org/display/JENKINS/Building+a+software+project#Buildingasoftwareproject-JenkinsSetEnvironmentVariables)

Are we sure that CHANGE_ID will never contain special charachter like '-' ?

**Comment by olblak** [comment]
https://wiki.jenkins-ci.org/display/JENKINS/Building+a+software+project#Buildingasoftwareproject-JenkinsSetEnvironmentVariables

**Comment by olblak** [comment]
[Doc](https://wiki.jenkins-ci.org/display/JENKINS/Building+a+software+project#Buildingasoftwareproject-JenkinsSetEnvironmentVariables)

**Comment by rtyler** [comment]
@olblak `CHANGE_ID` is provided by the GitHub Branch Source plugin, that wiki page is woefully out of date and doesn't consider plugin-provided environment variable.

**Pull request merged**